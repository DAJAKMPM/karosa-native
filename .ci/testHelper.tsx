import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
