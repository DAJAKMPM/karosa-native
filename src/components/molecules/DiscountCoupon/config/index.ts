/**
 *
 * @format
 *
 */

export const ICON = {
  HEIGHT: 88,
  WIDTH: 239,
};

export const COUPON_OPACITY = 0.9;
