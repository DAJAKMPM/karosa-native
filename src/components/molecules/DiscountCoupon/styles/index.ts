/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

const DiscountCouponStyles = StyleSheet.create({
  container: {
    alignContent: "center",
    justifyContent: "center",
    position: "relative",
    overflow: "hidden",
  },
  discountContainer: {
    position: "absolute",
    flexWrap: "wrap",
    left: "9%",
  },
  txtDiscount: {
    ...theme.textTitle,
    color: theme.colors.primary,
    textAlign: "center",
    fontWeight: "500",
  },
  detailsContainer: {
    position: "absolute",
    left: "42%",
    top: "20%",
    flexDirection: "column",
  },
  txtDetailsLight: {
    ...theme.textLight,
    color: theme.colors.dark20,
    fontWeight: "300",
  },
  txtMinSpend: {
    ...theme.textLightBold,
    color: theme.colors.dark20,
    fontWeight: "700",
  },
  bottomContainer: {
    flexDirection: "row",
  },
  btnMainContainer: {
    width: 49,
    position: "relative",
    bottom: 15,
    left: 5,
  },
  btnSubContainer: {
    height: 32,
    borderRadius: 3,
  },
  txtButton: {
    ...theme.textLight,
    textAlign: "center",
    fontWeight: "700",
  },
});

export default DiscountCouponStyles;
