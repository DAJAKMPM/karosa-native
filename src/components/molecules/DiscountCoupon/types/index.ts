/**
 *
 * @format
 *
 */

export type PropsType = {
  onCoupon?(): void;
  onClaim?(): void;
  discountPrice: string;
  minSpendPrice: string;
  validDate: string;
};
