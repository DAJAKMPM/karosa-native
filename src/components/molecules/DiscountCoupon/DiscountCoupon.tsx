/**
 *
 * DiscountCoupon
 * @format
 *
 */

import React, { FC } from "react";
import Icon from "@app/atoms/Icon";
import Text from "@app/atoms/Text";
import { TouchableOpacity, View } from "react-native";
import Button from "@app/atoms/Button";

import type { PropsType } from "./types";
import { COUPON_OPACITY, ICON } from "./config";
import DiscountCouponStyles from "./styles";

const DiscountCoupon: FC<PropsType> = (props) => {
  const { discountPrice, minSpendPrice, validDate, onClaim, onCoupon } = props;

  return (
    <TouchableOpacity
      onPress={onCoupon}
      activeOpacity={COUPON_OPACITY}
      style={DiscountCouponStyles.container}>
      <Icon
        group="products"
        name="coupon"
        height={ICON.HEIGHT}
        width={ICON.WIDTH}
      />
      <View style={DiscountCouponStyles.discountContainer}>
        <Text
          text={`P${discountPrice}`}
          textStyle={DiscountCouponStyles.txtDiscount}
        />
        <Text text="OFF" textStyle={DiscountCouponStyles.txtDiscount} />
      </View>
      <View style={DiscountCouponStyles.detailsContainer}>
        <Text
          text="Minimum Spend"
          textStyle={DiscountCouponStyles.txtDetailsLight}
        />
        <Text
          text={`P${minSpendPrice}`}
          textStyle={DiscountCouponStyles.txtMinSpend}
        />
        <Text
          text="Valid until:"
          textStyle={DiscountCouponStyles.txtDetailsLight}
        />
        <View style={DiscountCouponStyles.bottomContainer}>
          <Text
            text={validDate}
            textStyle={DiscountCouponStyles.txtDetailsLight}
          />
          <View style={DiscountCouponStyles.btnMainContainer}>
            <Button
              onPress={onClaim}
              titleStyle={DiscountCouponStyles.txtButton}
              buttonStyle={DiscountCouponStyles.btnSubContainer}
              title="Claim"
            />
          </View>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default DiscountCoupon;
