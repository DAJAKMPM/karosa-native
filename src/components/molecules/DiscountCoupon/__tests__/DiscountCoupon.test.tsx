import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import DiscountCoupon from "../DiscountCoupon";

describe("DiscountCoupon Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <DiscountCoupon
        minSpendPrice="200"
        discountPrice="200"
        validDate="Nov-20-2021"
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
