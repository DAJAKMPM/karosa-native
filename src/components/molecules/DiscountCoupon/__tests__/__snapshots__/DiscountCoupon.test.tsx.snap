// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountCoupon Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "justifyContent": "center",
        "opacity": 1,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <View>
      <SvgMock
        height={88}
        width={239}
      />
    </View>
    <View
      style={
        Object {
          "flexWrap": "wrap",
          "left": "9%",
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#0AA351",
            "fontFamily": "SFProText-Bold",
            "fontSize": 24,
            "fontWeight": "500",
            "lineHeight": 28,
            "textAlign": "center",
          }
        }
      >
        P200
      </Text>
      <Text
        style={
          Object {
            "color": "#0AA351",
            "fontFamily": "SFProText-Bold",
            "fontSize": 24,
            "fontWeight": "500",
            "lineHeight": 28,
            "textAlign": "center",
          }
        }
      >
        OFF
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "left": "42%",
          "position": "absolute",
          "top": "20%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1C1C1C",
            "fontFamily": "SFProText-Regular",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 14.06,
          }
        }
      >
        Minimum Spend
      </Text>
      <Text
        style={
          Object {
            "color": "#1C1C1C",
            "fontFamily": "SFProText-Bold",
            "fontSize": 12,
            "fontWeight": "700",
            "lineHeight": 14.06,
          }
        }
      >
        P200
      </Text>
      <Text
        style={
          Object {
            "color": "#1C1C1C",
            "fontFamily": "SFProText-Regular",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 14.06,
          }
        }
      >
        Valid until:
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1C1C1C",
              "fontFamily": "SFProText-Regular",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 14.06,
            }
          }
        >
          Nov-20-2021
        </Text>
        <View
          style={
            Object {
              "bottom": 15,
              "left": 5,
              "position": "relative",
              "width": 49,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                undefined,
                Object {
                  "backgroundColor": "#fff",
                  "overflow": "visible",
                  "shadowColor": "rgba(0,0,0, .4)",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 1,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#0AA351",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 32,
                    "justifyContent": "center",
                    "overflow": "hidden",
                    "padding": 16,
                    "position": "relative",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 12,
                      "fontWeight": "700",
                      "lineHeight": 14.06,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Claim
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
